src/ansible_compat/runtime.py

```diff
@@ -231,6 +235,11 @@ def warning(
         # Monkey patch ansible warning in order to use warnings module.
         Display.warning = warning
 
+    def _add_sys_path_to_collection_paths(self) -> None:
+        """Add the sys.path to the collection paths."""
+        if not self.isolated and self.config.collections_scan_sys_path:
```

Comment:
Since molecule calls with isolated False (https://github.com/ansible/molecule/blob/33e5d8bffb248c1013765e8347bb914d686d8038/src/molecule/app.py#L10) I wanted to keep the scope of the change to a minimum not know _exactly_ where isloated and why it might be used.

lint inits with "isolated=True" and I didn't want to impact it https://github.com/ansible/ansible-lint/blob/225366aa87636693e907204eeeba8cc1bd76e031/src/ansiblelint/app.py#L49

Maybe @ssbarnea has an opinion here?

Reply:
Here is my concern

For`sys.path`, as `ansible-galaxy` is invoked underneath, and only the path in `sys.path` contains a folder named `ansible_collections` is added in to `ansible-galaxy` search path.

To be exact, if `xxx/site-packages/ansible_collections` folder exists, we just need add `xxx/site-packages`. Otherwise, no `sys.path` should be added.

Besides `xxx/site-packages`, there are also some other path in sys.path, but I don't believe `ansible_collections` folder exists in the rest.

Please note `AnsibleCollectionConfig.collection_paths` in 
https://github.com/ansible/ansible/blob/863e2571db5d70b03478fd18efef15c4bde88c10/lib/ansible/cli/galaxy.py#L1647-L1649


https://github.com/ansible/ansible/blob/863e2571db5d70b03478fd18efef15c4bde88c10/lib/ansible/utils/collection_loader/_collection_finder.py#L254-L272